### DISCUSSION TITLE
Turn bridge-style into a proper Python package

### DISCUSSION THREAD

**Discussion by GeoSander** [GeoCat/bridge-style#109] [Ideas]
Having a release on pypi would be great. To get there the imports need to change a little, but it would be very beneficial. For the user it shouldn't change at all.

Even if no release is published to pypi if the module system is adjusted as a proper python package then it would be possible for users to use the egg syntax to install this instead of using a git submodule, which is quite a terrible way of having a dependency:

`pip install "git+https://github.com/GeoCat/bridge-style.git@519f8d2#egg=bridgestyle"`

_Originally posted by @categulario in https://github.com/GeoCat/bridge-style/issues/47#issuecomment-1258304406_

**Comment by GeoSander** [comment]
Hi @categulario, you are absolutely right and this is definitely something we should do.

Refactoring the repo as a proper package so you can properly use the egg syntax would be the preferred way for me, because I hate publishing something to PyPI that does not fully work out-of-the-box (i.e. because of the QGIS runtime dependency). However, we could make it super clear in the docs that QGIS/QML style conversions are not supported without a QGIS runtime of course.

Feel free to create a PR for this by the way, if you are bored. 🙂
I am occupied with other things at the moment...

**Comment by geraldo** [comment]
Packaging bridge-style as a proper python package would be great and would open the possibility to use it as dependency (for example from [qgis-bridge-plugin](https://github.com/GeoCat/qgis-bridge-plugin/)) instead of shipping the code. A nice way to do so is using [qpip](https://github.com/opengisch/qpip), which is a python dependency manager...
